### ISSUE TITLE
p:commandLink: display is not equal to <h:commandLink> when using value AND icon

### ISSUE THREAD

**Issue by schlebe** [primefaces/primefaces#3873]
## 1) Environment
- PrimeFaces version: 6.1.17, 6.3-SNAPSHOT
- Does it work on the newest released PrimeFaces version? NO
- Does it work on the newest sources in GitHub? NO
- Application server + version: Glassfish 4.3
- Affected browsers: Chrome / Internet Explorer 11

## 2) Expected behavior

When <p:commandLink> with value attribute and <i> tag (for icon) is displayed; only value is displayed.
If I use <h:commandLink>, value and icon are displayed.
I'm waiting that value + icon are displayed as original JSF widget.

## 3) Actual behavior

Only value/link/text attribute is displayed

## 5) Sample XHTML
```
<h:commandLink value="h:commandLink">
    <i class="fa fa-fw fa-chevron-circle-up"/>
</h:commandLink>
```
```
<p:commandLink value="p:commandLink">
    <i class="fa fa-fw fa-chevron-circle-up"/>
</p:commandLink>
```
## 6) Sample bean

It is only a display problem. No bean are necessary.

## 7) SOLUTION

suppress "else" in CommandLinkRenderer.java source at #131 and #151 lines

```
if (label != null) {
    writer.writeText(label, "value");
} else {
    renderChildren(context, link);
}
```

**Comment by melloware** [comment]
Attached is a sample you can run with "mvn clean jetty:run" and navigate to http://localhost:8080/primefaces-test/test.xhtml.

@schlebe  I can't reproduce here is what my sample displays...

![image](https://user-images.githubusercontent.com/4399574/42759108-cfb5aade-88d3-11e8-8546-2e4f542a9636.png)

**Comment by melloware** [comment]
OK I was able to replicate it.

![image](https://user-images.githubusercontent.com/4399574/42771295-b788625a-88f4-11e8-9bec-189c09c4419f.png)

Attached is the source:  
[primefaces-3873.zip](https://github.com/primefaces/primefaces/files/2198461/primefaces-3873.zip)

However getting rid of renderChildren is not the correct solution I don't think.

**Comment by schlebe** [comment]
Ok, you have same display as mine.

But I think that what I propose is the good solution. 

It is not the best solution because <h:commandLink> as a little issue : it display an empty underscored space character between VALUE attribute and icon. I have tried in replacing VALUE by a <span> at same level than <i> but I have observed same problem.

In fact, I have tried a lot of combinaison and only one work correctly.
```
<h:commandLink>
    <span>h:commandLink</span><i class="fa fa-fw fa-chevron-circle-up"/>
</h:commandLink>
```
The <i> element MUST BE just after </span> element. If <i> element is on a new line just behind <span> element, an empty UNDERSCORED character is displayed.

I have tried the same thing using <p:commandLink> but I have seen a little imperfection : a UNDERSCORE character is displayed just ... BEFORE text !            
```
<p:commandLink>
    <span>p:commandLink</span><i class="fa fa-fw fa-chevron-circle-up"/>
</p:commandLink>
```
I hope that all these explanations can help you to correct this little issue :-)

I have tested again in putting all component in only one line like this

 `<p:commandLink><span>p:commandLink</span><i class="fa fa-fw fa-chevron-circle-up"/></p:commandLink>
`

... and surprise, the first UNDERSCORED space-character has been removed !
